cylc/flow/xtriggers/workflow_state.py

```diff
@@ -14,101 +14,93 @@
-
-from cylc.flow.cycling.util import add_offset
-from cylc.flow.dbstatecheck import CylcWorkflowDBChecker
-from cylc.flow.pathutil import get_cylc_run_dir
-from cylc.flow.workflow_files import infer_latest_run_from_id
+from cylc.flow.scripts.workflow_state import WorkflowPoller
+from cylc.flow.id import tokenise
+from cylc.flow.exceptions import WorkflowConfigError
+from cylc.flow.task_state import TASK_STATUS_SUCCEEDED
 
 
 def workflow_state(
-    workflow: str,
-    task: str,
-    point: str,
+    workflow_task_id: str,
     offset: Optional[str] = None,
-    status: str = 'succeeded',
-    message: Optional[str] = None,
-    cylc_run_dir: Optional[str] = None
-) -> Tuple[bool, Optional[Dict[str, Optional[str]]]]:
-    """Connect to a workflow DB and query the requested task state.
+    flow_num: Optional[int] = 1,
+    alt_cylc_run_dir: Optional[str] = None,
+) -> Tuple[bool, Dict[str, Optional[str]]]:
+    """Connect to a workflow DB and check a task status or output.
 
-    * Reports satisfied only if the remote workflow state has been achieved.
-    * Returns all workflow state args to pass on to triggering tasks.
+    If the status or output has been achieved, return {True, result}.
 
     Arguments:
-        workflow:
-            The workflow to interrogate.
-        task:
-            The name of the task to query.
-        point:
-            The cycle point.
+        workflow_task_id:
+            ID (workflow//point/task:selector) of the target task.
         offset:
-            The offset between the cycle this xtrigger is used in and the one
-            it is querying for as an ISO8601 time duration.
-            e.g. PT1H (one hour).
-        status:
-            The task status required for this xtrigger to be satisfied.
-        message:
-            The custom task output required for this xtrigger to be satisfied.
-
-            .. note::
-
-               This cannot be specified in conjunction with ``status``.
-        cylc_run_dir:
+            Offset from cycle point as an ISO8601 or integer duration,
+            e.g. PT1H (1 hour) or P1 (1 integer cycle)
+        flow_num:
+            Flow number of the target task.
+        alt_cylc_run_dir:
             Alternate cylc-run directory, e.g. for another user.
 
             .. note::
 
-               This only needs to be supplied if the workflow is running in a
-               different location to what is specified in the global
-               configuration (usually ``~/cylc-run``).
+               This is only needed if the workflow is installed to a
+               non-standard location.
 
     Returns:
-        tuple: (satisfied, results)
-
+        tuple: (satisfied, result)
         satisfied:
             True if ``satisfied`` else ``False``.
-        results:
-            Dictionary containing the args / kwargs which were provided
-            to this xtrigger.
+        result:
+            Dict {workflow_id, task_id, task_selector, flow_number}.
```

Comment:
Explanation, if type hints are not present, then Google style docstrings should provide the type (followed by a colon) like so:

```
    Returns:
        bool: The return value. True for success, False otherwise.
```

However, when type hints are used, the type does not need to be specified:

```
    Returns:
        The return value. True for success, False otherwise.
```

Those examples are from this page of examples: https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html#example-google

Sphinx Napoleon will error when docstrings don't match its expected formats (either Google style or Numpy style).

Reply:
fixed